osu.Game/Screens/Play/Player.cs

```diff
@@ -81,6 +81,10 @@ public abstract class Player : ScreenWithBeatmapBackground, ISamplePlaybackDisab
 
         private bool isRestarting;
 
+        public bool RestartedViaHotkey;
+
+        public Bindable<bool> IsRestartingFromHotkey = new Bindable<bool>();
```

Comment:
Both of these are very haphazard. There's no definition of when they need to be set, nor what they do.

Is there a reason this can't be added to `PlayerConfiguration` as a `bool AutomaticallySkipIntro` or similar? That will define that it needs to be set at point of `Player` construction, and also better explain what it's doing.

Reply:
@BlauFx just as a heads up, we prefer you don't resolve PR feedback as it makes it very hard to track things (we will do that after confirming the issue has been resolved).